documentation updates (#951)
elastic/apm-agent-go: docs/log-correlation.asciidoc

}
----

Follow this article to ingest JSON-encoded logs with Filebeat:
{blog-ref}how-to-instrument-your-go-app-with-the-elastic-apm-go-agent#logs[How to instrument your Go app with the Elastic APM Go agent].

[float]
[[log-correlation-manual-unstructured]]
==== Manual log correlation (unstructured)

For correlating unstructured logs (e.g. basic printf-style logging, like the standard library's
`log` package), then you will need to need to include the trace IDs in your log message. Then,
extract them using Filebeat.

> If you already have a transaction and/or span object, then use the

Reviewer: ```suggestion
If you already have a transaction or span object, use the
```

Reviewer: If users have both, they will still be able to figure out how to do this.

> <<transaction-tracecontext, Transaction.TraceContext>> and <<span-tracecontext, Span.TraceContext>>

Reviewer: ```suggestion
<<transaction-tracecontext, Transaction.TraceContext>> or <<span-tracecontext, Span.TraceContext>>
```

    2019/09/17 14:48:02 ERROR [trace.id=cd04f33b9c0c35ae8abe77e799f126b7 transaction.id=cd04f33b9c0c35ae span.id=960834f4538880a4] an error occurred

> For the log correlation to work, the trace IDs must be extracted from the log message and

Reviewer: log correlation isn't usually preceded by an article.

Author: Reverted

> For the log correlation to work, the trace IDs must be extracted from the log message and

Reviewer: ```suggestion
For log correlation to work, the trace IDs must be extracted from the log message and
```

stored in separate fields in the Elasticsearch document. This can be achieved by

> {filebeat-ref}/configuring-ingest-node.html[parsing the data using an ingest node], in particular

Reviewer: ```suggestion
{filebeat-ref}/configuring-ingest-node.html[using an ingest pipeline to parse the data], in particular
```

Reviewer: I'd change the link. The topic title puts the parsing first to emphasize it, but you don't have to do that in the link.
Also I think we are getting away from talking about "ingest node" and instead using "ingest pipeline" because it's more intuitive.
